src/main/java/tr/com/srdc/cda2fhir/transform/CCDTransformerImpl.java

```diff
@@ -92,122 +92,189 @@ public CCDTransformerImpl() {
-     * @return A FHIR Bundle that contains a Composition corresponding to the CCD document and all other resources that are referenced within the Composition.
-     * @throws Exception 
-     */
-    public Bundle transformDocument(String filePath) throws Exception {
-    	ContinuityOfCareDocument cda = getClinicalDocument(filePath);
-    	return transformDocument(cda, true);
-    }
-    
-    /**
-     * Transforms a Consolidated CDA (C-CDA) 2.1 Continuity of Care Document (CCD) instance to a Bundle of corresponding FHIR resources
-     * @param cda A Consolidated CDA (C-CDA) 2.1 Continuity of Care Document (CCD) instance to be transformed
-     * @return A FHIR Bundle that contains a Composition corresponding to the CCD document and all other resources that are referenced within the Composition.
-     */
-    public Bundle transformDocument(ContinuityOfCareDocument cda) {
-    	return transformDocument(cda, true);
-    }
-    
+	}
+
+	/**
+	 * Transforms a Consolidated CDA (C-CDA) 2.1 Continuity of Care Document (CCD)
+	 * instance to a Bundle of corresponding FHIR resources
+	 * 
+	 * @param filePath A file path string to a Consolidated CDA (C-CDA) 2.1
+	 *                 Continuity of Care Document (CCD) on file system
+	 * @return A FHIR Bundle that contains a Composition corresponding to the CCD
+	 *         document and all other resources that are referenced within the
+	 *         Composition.
+	 * @throws Exception
+	 */
+	public Bundle transformDocument(String filePath) throws Exception {
+		ContinuityOfCareDocument cda = getClinicalDocument(filePath);
+		return transformDocument(cda, true);
+	}
+
+	/**
+	 * Transforms a Consolidated CDA (C-CDA) 2.1 Continuity of Care Document (CCD)
+	 * instance to a Bundle of corresponding FHIR resources
+	 * 
+	 * @param cda                A Consolidated CDA (C-CDA) 2.1 Continuity of Care
+	 *                           Document (CCD) instance to be transformed
+	 * @param bundleType         The type of bundle to create, currently only
+	 *                           supports transaction bundles.
+	 * @param resourceProfileMap The mappings of default resource profiles to
+	 *                           desired resource profiles. Used to set profile
+	 *                           URI's of bundle entries or omit unwanted entries.
+	 * @param documentBody       The decoded base64 document that would be included
+	 *                           in the provenance object if provided.
+	 * @return A FHIR Bundle that contains a Composition corresponding to the CCD
+	 *         document and all other resources that are referenced within the
+	 *         Composition.
+	 * @throws Exception
+	 */
+	public Bundle transformDocument(ContinuityOfCareDocument cda, BundleType bundleType,
+			Map<String, String> resourceProfileMap, String documentBody, Identifier assemblerDevice) throws Exception {
```

Comment:
So I think we had another disconnect here. I'm pretty sure there is a function on the CDAUtil that lets you cast the ContinuityOfCareDocument as a file, and I was saying the output of that function is what we would consider to be document body for provenance purposes. That said, it's not perfect provenance since we would be re-creating the source for provenance purposes, so there's an argument you can make against it which is part of the reason I'm all over the map on this one. So yeah, I'm okay with this approach (I think).